src/eureka/S5_lightcurve_fitting/models/HSTRampModel.py

```diff
@@ -0,0 +1,105 @@
+import numpy as np
+
+from .Model import Model
+from ...lib.readEPF import Parameters
+
+
+class HSTRampModel(Model):
+    """Model for HST orbit-long exponential plus quadratic ramps"""
+    def __init__(self, **kwargs):
+        """Initialize the HST ramp model.
+
+        Parameters
+        ----------
+        **kwargs : dict
+            Additional parameters to pass to
+            eureka.S5_lightcurve_fitting.models.Model.__init__().
+            Can pass in the parameters, longparamlist, nchan, and
+            paramtitles arguments here.
+        """
+        # Inherit from Model class
+        super().__init__(**kwargs)
+
+        # Define model type (physical, systematic, other)
+        self.modeltype = 'systematic'
+
+        # Check for Parameters instance
+        self.parameters = kwargs.get('parameters')
+
+        # Generate parameters from kwargs if necessary
+        if self.parameters is None:
+            coeff_dict = kwargs.get('coeff_dict')
+            params = {hN: coeff for hN, coeff in coeff_dict.items()
+                      if hN.startswith('h') and hN[1:].isdigit()}
+            self.parameters = Parameters(**params)
+
+        # Set parameters for multi-channel fits
+        self.longparamlist = kwargs.get('longparamlist')
+        self.nchan = kwargs.get('nchan')
+        self.paramtitles = kwargs.get('paramtitles')
+
+        # Update coefficients
+        self.coeffs = np.zeros((self.nchan, 6))
+        self._parse_coeffs()
+
+    @property
+    def time(self):
+        """A getter for the time."""
+        return self._time
+
+    @time.setter
+    def time(self, time_array):
+        """A setter for the time."""
+        self._time = time_array
+        if self.time is not None:
+            # Convert to local time
+            self.time_local = self.time - self.time[0]
```

Comment:
Take a look at the current PolynomialModel and ExpRampModel time setter functions. There's some added complexity with the recent addition of multwhite fits